src/configurations.jl

```diff
@@ -388,7 +388,7 @@ julia> c"[Kr] 4d10 5s2 4f2"s
 ```
 """
 macro c_str(conf_str, suffix="")
-    parse_conf_str(Orbital, conf_str, suffix)
+    :(parse_conf_str(Orbital, $conf_str, $suffix))
```

Comment:
Why turn all these into expressions? This will stop them from being evaluated at parse time, no?

Reply:
Since it means you shift the work from parse time to when it's actually needed. Say you have a lot of string macros in your code, then just loading the file will take a lot of time, even if they are stuck in a function that you don't execute directly.

The other, more important thing is that I discovered that you can't do `@test_throws ... ` on a string macro to check if it handles malformed strings correctly, if it is executed at parse time.

Reply:
In my view, generally, the point of a string macro is to shift computation to parse time (e.g. `@r_str`). If you want run-time, then you could just have a function. I think with this implementation you're more likely to run into suboptimal performance if you do something like

```julia
(conf == c"1s2") ? foo : bar
```

in a tight loop.

Reply:
Case in point, a common pattern that I use is to pass a keyword argument for `kappa` with a default value. Something like

```julia
function foo(; kappa = \kappa"s")
    ...
end
```

Reply:
No, I can't say I have :) for large amounts of orbitals/configurations, you are better off generating them programmatically.

I concede that we revert that commit if we can figure out how to fix
> The other, more important thing is that I discovered that you can't do @test_throws ... on a string macro to check if it handles malformed strings correctly, if it is executed at parse time.

Reply:
Sorry for bypassing review, btw. I felt this was uncomplicated, but it never is when you think it is :P

Reply:
The testing issue is a fair point. But it looks like it should be possible to work around that with `macroexpand`:

```julia
@test_throws LoadError macroexpand(@__MODULE__, :( rso"1s(0,1/2)" ))
```

It won't throw `ArgumentError` directly, instead it will be wrapped in a `LoadError`.

Apparently, theoretically you could actually test against an exact error object, because this works

```julia
@test_throws LoadError(@__FILE__, @__LINE__, ArgumentError("...")) macroexpand(@__MODULE__, :( rso"1s(0,1/2)" ))
```

But you'd have to provide the exact string to `ArgumentError`. Either way, this is just for fun -- I don't think it's actually important to test the exact error type here.

---

Coming back to the parse vs run-time just quickly: if it ever becomes necessary, we can always add corresponding functions for each string macro (so e.g. `rso"1s(0,1/2)"` vs `rso("1s(0,1/2)")`).